Feature/lumina node integration 2

gabros20 opened the pull request.

jcstein reviewed (approved):
lgtm, leaving comments for a small change

jcstein commented on src/pages/markdown-pages/learn/creating-new-blockchains.md:
```diff
@@ -13,7 +13,7 @@ description: "How modular blockchains can change the process of launching and de
   <meta name="twitter:title" content="Modular Scalability">
   <meta name="twitter:description" content="Traditionally, creating a new blockchain had been a resource-intensive endeavor, imposing considerable time and monetary costs. Although blockchains can be implemented more efficiently now, existing hurdles are still present that cause friction in the bootstrapping process.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/learn-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">
```

jcstein commented on src/pages/markdown-pages/learn/first-principles-of-modular-blockchains.md:
```diff
@@ -14,7 +14,7 @@ description: "A first principles perspective on modular blockchains."
   <meta name="twitter:title" content="First Principles of Modular Blockchains">
   <meta name="twitter:description" content="By beginning with the first principles of modular blockchains, their purpose and goals will become easier to understand.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/learn-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">

```

jcstein commented on src/pages/markdown-pages/learn/basics-of-modular-blockchains.md:
```diff
@@ -12,11 +12,12 @@ description: "What are modular blockchains and what makes them different to thei
+  <meta name="twitter:title" content="Basics of Modular Blockchains">
   <meta name="twitter:description" content="Blockchain researchers have long grappled with the challenge of creating the optimal system. Many architectures have been tried whose goal was to accommodate all users on a single chain or a tightly coupled group of chains that live under a single network.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/learn-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">
```

jcstein commented on src/pages/markdown-pages/learn/modular-architectures.md:
```diff
@@ -14,7 +14,7 @@ description: "A look into the different ways in which modular blockchains can be
   <meta name="twitter:title" content="Modular Scalability">
   <meta name="twitter:description" content="A new paradigm is emerging in which modular blockchains are enabling new chains to be constructed in ways that were not previously possible.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/learn-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">
```

jcstein commented on src/pages/markdown-pages/learn/modular-scalability.md:
```diff
@@ -14,7 +14,7 @@ description: "The scalability properties of different types of modular blockchai
   <meta name="twitter:title" content="Modular Scalability">
   <meta name="twitter:description" content="Monolithic blockchains have pursued varying scalability techniques over time, many of which have proven unsuccessful. Modular blockchains present a way for a monolithic blockchain to be split up among multiple specialized layers.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/learn-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">
```

jcstein commented on src/pages/markdown-pages/learn/modular-settlement-layers.md:
```diff
@@ -14,7 +14,7 @@ description: "Settlement layers in the modular paradigm."
   <meta name="twitter:title" content="Modular Settlement Layers">
   <meta name="twitter:description" content="By beginning with the first principles of modular blockchains, their purpose and goals will become easier to understand.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/learn-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">

```

jcstein commented on src/pages/markdown-pages/learn/values-of-modular-blockchains.md:
```diff
@@ -14,7 +14,7 @@ description: "The values that modular blockchains aim for."
   <meta name="twitter:title" content="Values of Modular Blockchains">
   <meta name="twitter:description" content="Blockchains are a social movement as much as they are a technology. Without values, blockchains would be no better than web2 tech companies. Here are the values that modular blockchains aim for.">
-  <meta name="twitter:image" content="https://raw.githubusercontent.com/alex-beckett/celestia.org/values-of-modular-article-addition/src/pages/markdown-pages/learn/images/values-of-modular-twitter-card.png">
+  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card%201.png">
```
```suggestion
  <meta name="twitter:image" content="https://raw.githubusercontent.com/celestiaorg/celestia.org/main/src/pages/markdown-pages/learn/images/Learn_Modular_Twitter_Card.png">

```

jcstein reviewed (changes requested):
lgtm, could you remove ` 1` from the file name?

gabros20 committed:
feat: Add NodeStatus component and related SVG icons

- Created NodeStatus component for displaying block verification status
- Added LuminaCheckmarkSVG, LuminaDiagonalArrowSVG, LuminaErrorSVG, and LuminaGradientCircleSVG
- Integrated NodeStatus into DesktopNav
- Updated .gitignore to exclude docs/ directory

gabros20 committed:
chore: Update Next.js config and SVG components

- Add Next.js configuration with page extensions, image optimization, and redirects
- Fix SVG components by updating SVG attribute syntax (stroke-* to strokeLinecap/strokeLinejoin)
- Migrate inline styles to React object syntax in LuminaGradientCircleSVG
- Sanitize JSON-like attributes in SVG components

gabros20 committed:
feat: working connection

gabros20 committed:
fix: loader svg animation direction

gabros20 committed:
fix: updated loader svg

gabros20 committed:
fix: multiple hydration issues in the original project build

gabros20 committed:
feat: Enhance NodeStatus component with dynamic syncing progress and improved status messaging

gabros20 committed:
chore: Refactor blog fetching, NodeStatus, and navigation components

- Improve blog post fetching with error handling and caching
- Update NodeStatus component styling and progress display
- Relocate AutoConnectingLuminaNode component
- Simplify LuminaDiagonalArrowSVG icon

gabros20 committed:
feat: Enhance animation and navigation components

- Refactor MovingGradient component to generate random gradient animations on client-side
- Update AutoConnectingLuminaNode to manage initialization state
- Modify NodeStatus to include animation controls and improved syncing messages
- Introduce MenuButton component for mobile navigation with SVG icons
- Clean up Nav component to integrate MenuButton and streamline layout

gabros20 committed:
fix: wasm issues build time

gabros20 committed:
feat: Update NodeStatus and useLuminaNode for improved syncing display

- Refactor NodeStatus to utilize syncProgress for approximate syncing percentage display as much possible
- Remove fake progress logic and simplify state management
- Enhance useLuminaNode hook to calculate and provide sync progress and connected peers
- Update event handling for better status updates during syncing and connection

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
Update package-lock.json and EcosystemExplorer.js: bump package versions and clean up imports

gabros20 committed:
fix: connection issues

gabros20 committed:
fix: inifinie loop

gabros20 committed:
fix loading state indication

gabros20 committed:
feat: working block loading

gabros20 committed:
feat: new block rendering component added

gabros20 committed:
Merge branch 'feature/lumina-node-v2' into feature/lumina-node-integration

gabros20 committed:
commdent

gabros20 committed:
fix: live block render

gabros20 committed:
fix: build error

gabros20 committed:
Merge branch 'feature/test-lumina-node' into feature/lumina-node-integration

gabros20 committed:
Merge branch 'main' into feature/lumina-node-integration

gabros20 committed:
fix: animations

gabros20 committed:
feat: add sync percentage display and improve block number handling

gabros20 committed:
refactor: show block number with sync indicator

gabros20 committed:
feat: add circular progress indicator for syncing status

gabros20 committed:
fix: mobile element sizes

gabros20 committed:
feat: add manual start/stop controls for Lumina light node sync

- Add start/stop buttons with manual sync control instead of auto-initialization

- Create LuminaStartSVG (green play) and LuminaStopSVG (red X) components

- Update AutoLuminaContext to support manual node control with startNode/stopNode functions

- Modify useLuminaNode hook to start with 'idle' state and manual start/stop functionality

- Add start button in idle state with 'Start Light Node' text (no spinning icon)

- Add stop button during syncing phase (only when block number available)

- Preserve existing CircularProgressIndicator for sync percentage display

- Update button positioning and spacing to prevent UI conflicts

- Ensure proper button priority: stop button during sync, explorer link when complete

- Add proper cleanup of event listeners and polling when sync is stopped

- Maintain responsive design with buttons hidden on mobile

- Fix button timing to prevent early appearance during initialization

Breaking changes:

- Node no longer auto-starts syncing on load, requires user interaction

- shouldInitialize prop removed from AutoLuminaContextProvider

gabros20 committed:
feat: make start/stop buttons visible on mobile with responsive sizing

- Remove hidden class from start/stop buttons to show on mobile
- Add responsive button sizing: 28px mobile, 36px desktop
- Add responsive icon scaling: 75% mobile, 100% desktop
- Update mobile width calculations for button visibility
 Adjust margins for smaller mobile button sizes - Reduce idle state widths: 135px mobile, 204px desktop
